cmd/exec_unix.go

```diff
@@ -23,3 +27,31 @@ func exec(ctx context.Context, command string, args []string, env []string) erro
 	// Only return if the execution fails.
 	return errors.Wrap(syscall.Exec(argv0, argv, env), "error executing command")
 }
+
+func getAWSEnvVars(assumeRoleOutput *sts.AssumeRoleWithWebIdentityOutput, awsOIDCConfig *aws_config_client.AWSOIDCConfiguration) []string {
+
+	// Load config profile values if those environment variables don't exist (lowest precedence)
+	envVars := []string{}
+	_, present := os.LookupEnv("AWS_REGION")
+	if (awsOIDCConfig.Region != "") && !present {
+		envVars = append(envVars, fmt.Sprintf("AWS_REGION=%s", awsOIDCConfig.Region))
+	}
+	_, present = os.LookupEnv("AWS_OUTPUT")
+	if awsOIDCConfig.Output != "" && !present {
+		envVars = append(envVars, fmt.Sprintf("AWS_OUTPUT=%s", awsOIDCConfig.Output))
+	}
+
+	// Load assumeRoleOutput credentials
+	envVars = append(envVars,
+		fmt.Sprintf("AWS_ACCESS_KEY_ID=%s", string(*assumeRoleOutput.Credentials.AccessKeyId)),
+		fmt.Sprintf("AWS_SECRET_ACCESS_KEY=%s", string(*assumeRoleOutput.Credentials.SecretAccessKey)),
+		fmt.Sprintf("AWS_SESSION_TOKEN=%s", string(*assumeRoleOutput.Credentials.SessionToken)),
+	)
+
+	// our environment variables take precedence
+	envVars = append(envVars,
+		os.Environ()...,
+	)
```

Comment:
I wonder if we should only do this for the `exec` command and not the env command. (the semantics we had before)

Reply:
Huh, what semantics?

Reply:
left a comment above about only exec being the one that gets all the env variables while `env` only gets the aws specific ones. That was the behavior we had before so I'd rather not change that with this pr (to avoid adding unrelated complexity) to this fix.